runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheServerFetchFactory.java

```diff
@@ -1187,15 +1187,16 @@ private void onServerFanoutInitialSignalSegmentDelete(
             final long now = currentTimeMillis();
 
             Node segmentNode = partition.sentinel().next();
-            while (segmentNode != partition.head() &&
-                    partition.deleteAt(segmentNode.segment(), retentionMillisMax) <= now)
+            while (segmentNode.segment() != null &&
```

Comment:
In general, the `Node` constructors enforce either that the `segment` is non-null, or it is the `sentinel` for the partition.

If we agree on that, then let's use `!node.sentinel()` instead of `node.segment() != null`.